runtime/caching/cache.go

```diff
@@ -111,9 +116,14 @@ func (c *Cache) Require(ctx Context, options interface{}) (*Handle, error) {
 		c.entries = append(c.entries, entry)
 
 		go entry.created.Do(func() {
-			defer entry.ctx.dispose() // ensure resources are cleanup when constructor is done
+			defer entry.ctx.dispose()
+			// ensure resources are cleanup when constructor is done
```

Comment:
This comment pertains to the line of the code that it follows? that seems unwise.

Also some grammar (which was probably there before), but `are cleaned up`